docs/content/best-practices/modeling.mdx

```diff
@@ -0,0 +1,176 @@
+      define parent: [resource]
+      
+      define editor : [role#assignee] or editor from entity or editor from parent
+      define viewer : [role#assignee] or editor or viewer from parent
+```
+
+We do not recommend this approach. Instead, you should create models that mimic as closely as possible the business logic of the application instead of generic models. The rule of thumb is that **if the end-user of the application can define certain entities/relations, then those should be represented in tuples. If not, it should be represented in the model**.
+
+For example, when defining roles, in general you have certain roles that come built-in with the application like an “admin” or a “billing manager”. The way you would model that is very simple:
+
+```dsl.openfga
+
+  type user
+
+  type entity
+    relations
+      define admin: [user]
+      define billing_manager: [user]
+
+      define can_manage_billing: admin or billing_manager
+      define can_manage_users: admin
+```
+
+Defining the model in a way that closely resembles your application domain has several advantages:
+
+  - **Enhanced clarity and maintainability**: Authorization logic is easier to understand, debug, and maintain. By just reading the model, developers or security auditors can readily grasp the meaning of each type and relationship.
+
+  - **Better performance**: Models with more specific types and flatter hierarchies often lead to better performance. This is because <ProductName format={ProductNameFormat.ShortForm}/> can more efficiently process queries involving well-defined types and relationships compared to navigating complex, high-cardinality recursive relationships within a single generic type.
+
+  - **Leveraging the model's flexibility**: <ProductName format={ProductNameFormat.ShortForm}/>'s modeling language is designed to be adaptable. You can define numerous distinct types and relationships without significant overhead. Model changes rarely require data migrations, allowing you to evolve your model as your application grows. <ProductName format={ProductNameFormat.ShortForm}/> is already a generic authorization engine, you don't need to build another one on top of it.
+
+  - **Improved Collaboration**: The resource types that are owned by each application team can be maintained in independent [modules](./../modeling/modular-models.mdx). You can then control which application can write to specific resource types. For example, when using the API credential issued to a Document Management application, you can only write/delete tuples for documents and folders, but not for other types. This is not possible if you use a generic `resource` type, for example. 
+
+## Custom Roles
+
+However, in some applications, you will want end-users to define their own roles. In that case, you do need a role type:
+
+```dsl.openfga
+ model
+    schema 1.1 
+ type user
+
+  type role
+   relations
+     define assignee: [user]
+
+  type entity
```

Comment:
Same as below, any reason we are using entity instead of  application?

Reply:
To me I worry that we're encouraging vague type names that people can use for muliple perposes